### PR TITLE
Check the package name on core/ledger/kvledger UT (#2987)

### PR DESCRIPTION
**Pull request by justin-themedium**
Check the package name not local file name including "fabric/"
on core/ledger/kvledger unit test

`core/ledger/kvledger` unit-test has been failed on a directory that not "fabric".

Signed-off-by: Justin Yang <justin.yang@themedium.io>

#### Type of change
- Test update

#### Description
Check the error message without "`fabric/`" in `core/ledger/kvledger/snapshot_test.go`
When you run the unit-test in a directory that is not "`fabric`", that unit-test will fail because of checking the filename starting "`fabric/`".

#### Additional details
* Problem
`core/ledger/kvledger/` unit test fails when running the unit-test in another directory, not `fabric/`

```
--- FAIL: TestGenerateSnapshotErrors (13.43s)
    --- FAIL: TestGenerateSnapshotErrors/block_store_returns_error (1.10s)
        snapshot_test.go:571:
            	Error Trace:	snapshot_test.go:571
            	Error:      	"leveldb: closed\ninternal leveldb error while obtaining db iterator\ngithub.com/hyperledger/fabric/common/ledger/util/leveldbhelper.(*DBHandle).GetIterator\n\t/Users/medium/src/hyperledger/fabrictest/common/ledger/util/leveldbhelper/leveldb_provider.go:262\ngithub.com/hyperledger/fabric/common/ledger/blkstorage.(*blockIndex).exportUniqueTxIDs\n\t/Users/medium/src/hyperledger/fabrictest/common/ledger/blkstorage/blockindex.go:304\ngithub.com/hyperledger/fabric/common/ledger/blkstorage.(*BlockStore).ExportTxIds\n\t/Users/medium/src/hyperledger/fabrictest/common/ledger/blkstorage/blockstore.go:105\ngithub.com/hyperledger/fabric/core/ledger/kvledger.(*kvLedger).generateSnapshot\n\t/Users/medium/src/hyperledger/fabrictest/core/ledger/kvledger/snapshot.go:114\ngithub.com/hyperledger/fabric/core/ledger/kvledger.TestGenerateSnapshotErrors.func4\n\t/Users/medium/src/hyperledger/fabrictest/core/ledger/kvledger/snapshot_test.go:567\ntesting.tRunner\n\t/usr/local/Cellar/go@1.14/1.14.15/libexec/src/testing/testing.go:1050\nruntime.goexit\n\t/usr/local/Cellar/go@1.14/1.14.15/libexec/src/runtime/asm_amd64.s:1373" does not contain "fabric/common/ledger/blkstorage/blockindex.go"
            	Test:       	TestGenerateSnapshotErrors/block_store_returns_error
    --- FAIL: TestGenerateSnapshotErrors/config_history_mgr_returns_error (1.32s)
        snapshot_test.go:581:
            	Error Trace:	snapshot_test.go:581
            	Error:      	"leveldb: closed\ninternal leveldb error while obtaining db iterator\ngithub.com/hyperledger/fabric/common/ledger/util/leveldbhelper.(*DBHandle).GetIterator\n\t/Users/medium/src/hyperledger/fabrictest/common/ledger/util/leveldbhelper/leveldb_provider.go:262\ngithub.com/hyperledger/fabric/core/ledger/confighistory.(*db).getNamespaceIterator\n\t/Users/medium/src/hyperledger/fabrictest/core/ledger/confighistory/db_helper.go:114\ngithub.com/hyperledger/fabric/core/ledger/confighistory.(*Retriever).ExportConfigHistory\n\t/Users/medium/src/hyperledger/fabrictest/core/ledger/confighistory/mgr.go:221\ngithub.com/hyperledger/fabric/core/ledger/kvledger.(*kvLedger).generateSnapshot\n\t/Users/medium/src/hyperledger/fabrictest/core/ledger/kvledger/snapshot.go:120\ngithub.com/hyperledger/fabric/core/ledger/kvledger.TestGenerateSnapshotErrors.func5\n\t/Users/medium/src/hyperledger/fabrictest/core/ledger/kvledger/snapshot_test.go:577\ntesting.tRunner\n\t/usr/local/Cellar/go@1.14/1.14.15/libexec/src/testing/testing.go:1050\nruntime.goexit\n\t/usr/local/Cellar/go@1.14/1.14.15/libexec/src/runtime/asm_amd64.s:1373" does not contain "fabric/core/ledger/confighistory/mgr.go"
            	Test:       	TestGenerateSnapshotErrors/config_history_mgr_returns_error

(...)

FAIL
coverage: 78.8% of statements
FAIL	github.com/hyperledger/fabric/core/ledger/kvledger	583.226s
```

* Reproduce procedure
1. `$ git clone https://github.com/hyperledger/fabric.git fabrictest` or `$ mv fabric fabrictest`
2. `$ cd fabrictest`
3. `$ make unit-test`

### CODE REVIEW

**Comment by manish-sethi** [comment]
Why do you want to rename the folder? Fabric has always been following the practice of maintaining one-to-one mapping between the folder name and the golang package name.

**Comment by justin-themedium** [comment]
I just found it when I was comparing the source code between the local copies having different releases respectively and checking the build.
Should I keep the mapping between my local directory name and the golang package name?

It doesn't mean that I can use hlf, fabric-2.4-beta, awesome_fabric, good_job_fabric, super_fabric, etc, ... in my machine? ;)

**Comment by manish-sethi** [comment]
> I just found it when I was comparing the source code between the local copies having different releases respectively and checking the build. Should I keep the mapping between my local directory name and the golang package name?
> 
You can always compare git branches.

> It doesn't mean that I can use hlf, fabric-2.4-beta, awesome_fabric, good_job_fabric, super_fabric, etc, ... in my machine? ;)

If you want to go this way instead of different branches You can still do this by checking out in a higher level folder of these names :-)

**Comment by justin-themedium** [comment]
> > I just found it when I was comparing the source code between the local copies having different releases respectively and checking the build. Should I keep the mapping between my local directory name and the golang package name?
> 
> You can always compare git branches.
> 

I know this way but I just prefer that way that comparing between local copies. But the unit-test enforces the local directory name.

> > It doesn't mean that I can use hlf, fabric-2.4-beta, awesome_fabric, good_job_fabric, super_fabric, etc, ... in my machine? ;)
> 
> If you want to go this way instead of different branches You can still do this by checking out in a higher level folder of these names :-)

But if so, I will always face the build error because of this unit test when I run 'make'. I think it's okay that we can just check the relative path.

**Comment by manish-sethi** [comment]
> But if so, I will always face the build error because of this unit test when I run 'make'. I think it's okay that we can just check the relative path.

Not sure what you mean by this. If you have a folder `super_fabric/fabric`, why will it fail?

As such I don't have any problem in accepting this change as its minor change in a UT that does not break any tight condition that we want to preserve. However, in the future someone may again push another patch mentioning that the path is not complete and there won't be a reason not to accept that. 

In other words, I do not object to this change if this makes your life easier. So, can you rather make this change a bit differently -- In the UT replace the file name by the package name. E.g., `fabric/common/ledger/blkstorage/blockindex.go` -> `github.com/hyperledger/fabric/common/ledger/blkstorage`.

**Comment by justin-themedium** [comment]
> > But if so, I will always face the build error because of this unit test when I run 'make'. I think it's okay that we can just check the relative path.
> 
> Not sure what you mean by this. If you have a folder `super_fabric/fabric`, why will it fail?
 
In that way, it works. I don't think it's a good way, though.

> As such I don't have any problem in accepting this change as its minor change in a UT that does not break any tight condition that we want to preserve. However, in the future someone may again push another patch mentioning that the path is not complete and there won't be a reason not to accept that.
> 
> In other words, I do not object to this change if this makes your life easier. So, can you rather make this change a bit differently -- In the UT replace the file name by the package name. E.g., `fabric/common/ledger/blkstorage/blockindex.go` -> `github.com/hyperledger/fabric/common/ledger/blkstorage`.

I don't think it's worth that. Because It's more useful to show the local file name with the line number when you debug. The unit-test tests your local modifications.

**Comment by manish-sethi** [comment]
> I don't think it's worth that. Because It's more useful to show the local file name with the line number when you debug. The unit-test tests your local modifications.

In the UT, this line is not for printing file name/line number. The UTs are not supposed to fail and if they do anyways the error stack-trace will be printed. In this UT it's merely checking that the error is coming from the package expected so as to make sure that if the lower layer returns an error, it gets propagated to the top.

**Comment by justin-themedium** [comment]
> > I don't think it's worth that. Because It's more useful to show the local file name with the line number when you debug. The unit-test tests your local modifications.
> 
> In the UT, this line is not for printing file name/line number. The UTs are not supposed to fail and if they do anyways the error stack-trace will be printed. In this UT it's merely checking that the error is coming from the package expected so as to make sure that if the lower layer returns an error, it gets propagated to the top.

Yes, right. It shows the stack trace, but the stack trace will show you the local file name and line number.
Please see issue #2987. You can see the stack trace that contains "`/Users/medium/src/hyperledger/fabrictest/core/ledger/confighistory/mgr.go:221`".

and Yes, it's merely checking the error so I think it's enough to check the relative path like "`core/ledger/confighistory/mgr.go`" and this pull request is just for that.

**Comment by manish-sethi** [comment]
> Yes, right. It shows the stack trace, but the stack trace will show you the local file name and line number. Please see issue #2987. You can see the stack trace that contains "`/Users/medium/src/hyperledger/fabrictest/core/ledger/confighistory/mgr.go:221`".
> 
And the same stack trace contains the package name `github.com/hyperledger/fabric/core/ledger/confighistory`. Why not use something that does not change with install location?

**Comment by justin-themedium** [comment]
> > Yes, right. It shows the stack trace, but the stack trace will show you the local file name and line number. Please see issue #2987. You can see the stack trace that contains "`/Users/medium/src/hyperledger/fabrictest/core/ledger/confighistory/mgr.go:221`".
> 
> And the same stack trace contains the package name `github.com/hyperledger/fabric/core/ledger/confighistory`. Why not use something that does not change with install location?

I don't want to change the original author's purpose that checking the file name not the package name. Is it enough to check only the package name? if so, I will change that.

**Comment by manish-sethi** [comment]
> I don't want to change the original author's purpose that checking the file name not the package name. Is it enough to check only the package name? if so, I will change that.

Myself is the original author for this :-)  So yes, please go ahead with the package name change.

**Comment by justin-themedium** [comment]
> > I don't want to change the original author's purpose that checking the file name not the package name. Is it enough to check only the package name? if so, I will change that.
> 
> Myself is the original author for this :-) So yes, please go ahead with the package name change.

OMG! Okay! I will. Thank you!

**Review by manish-sethi — APPROVED** [review]
